castai/commitments/mapping.go

```diff
@@ -201,45 +251,86 @@ func MapCUDImportToResource(
 	return res, nil
 }
 
-// cudConfigMatcherKey is a utility type for mapping CUDs to their configurations
-type cudConfigMatcherKey struct {
+func MapReservationImportToResource(
+	cudWithCfg *commitmentWithConfig[CastaiAzureReservationImport],
+) *AzureReservationResource {
+	res := &AzureReservationResource{
+		Count:              int(lo.FromPtr(cudWithCfg.Commitment.Quantity)),
+		ReservationID:      lo.FromPtr(cudWithCfg.Commitment.ReservationId),
+		ReservationStatus:  lo.FromPtr(cudWithCfg.Commitment.Status),
+		StartTimestamp:     lo.FromPtr(cudWithCfg.Commitment.PurchaseDate),
+		EndTimestamp:       lo.FromPtr(cudWithCfg.Commitment.ExpirationDate),
+		Name:               lo.FromPtr(cudWithCfg.Commitment.Name),
+		Region:             lo.FromPtr(cudWithCfg.Commitment.Region),
+		InstanceType:       lo.FromPtr(cudWithCfg.Commitment.ProductName),
+		Plan:               lo.FromPtr(cudWithCfg.Commitment.Term),
+		Scope:              lo.FromPtr(cudWithCfg.Commitment.Scope),
+		ScopeResourceGroup: lo.FromPtr(cudWithCfg.Commitment.ScopeResourceGroup),
+		ScopeSubscription:  lo.FromPtr(cudWithCfg.Commitment.ScopeSubscription),
+	}
+
+	switch res.Plan { // normalize the values just like CAST AI's API does
+	case "P1Y":
+		res.Plan = "ONE_YEAR"
+	case "P3Y":
+		res.Plan = "THREE_YEAR"
+	case "ONE_YEAR":
+	case "THREE_YEAR":
+	default:
+		res.Plan = "ONE_YEAR"
```

Comment:
Do we need to have a default value? or return validation error?

Reply:
This is what the API does so my intention here was to mimic its behaviour: https://gitlab.com/castai/kubecast/blob/15a6e46e7e23680a1104de2f55a4812f3cb3911a/services/inventory/internal/server/commitments/server.go#L535

Now when I think of it, it might be good to return an error actually.

Reply:
please, return error